src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +152,63 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            if (_encoding == null || _decoder == null)
+            {
+                _encoding = Encoding.UTF8;
```

Comment:
There's no need for _encoding.  You can just use Encoding.UTF8 directly whenever you need it.